### ISSUE TITLE
RFC: InsertQuery implementation. PDO quote vs prepared statement

### ISSUE THREAD

**Issue by maldoinc** [lichtner/fluentpdo#130]
Is there any particular reason the `InsertQuery` class quotes the arguments by itself rather than using prepared statements? Prepared statements are more secure than manually quoing the values and possibly even faster.

Not to mention this makes working with Microsoft SQL Server, any ODBC interface and possibly other engines impossible due to:

* ODBC not supporting the `PDO::quote` method, as stated in the [PHP Manual](https://secure.php.net/manual/en/pdo.quote.php)
* SQL Server demands literal strings (which `InsertQuery` class generates) being prepended with `N` in order for them to be treated as unicode (eg: `N'this string'`). Without it the data is stored in the default database codepage (most of the time, Latin1 -- thus, gibberish as far as unicode is concerned).  Stated in: [Microsoft KB 239530](https://support.microsoft.com/en-us/kb/239530)

Both of these problems are solved by using prepared statements with parameters. Are there any drawbacks to this that I'm not seeing?

Thanks.

PS: 
To make working with SQL Server possible we are using a forked version which has this code before the final return of the `BaseQuery.quote` method. This is not entirely optimal as having prepared statements would be better, but this is the easier of the two modifications to implement.

```php
if ($this->fpdo->getPdo()->getAttribute(\PDO::ATTR_DRIVER_NAME) === 'sqlsrv') {
    return 'N' . $pdo->quote($value);
}
```

**Comment by jkufner** [comment]
I aggree with prepared statements. The bahavior should be consistent with selects and all other features.

Prepending 'N' is not very nice fix.

**Comment by xhlula** [comment]
I also agree with prepared statements. @maldoinc :thumbsup:

**Comment by maldoinc** [comment]
The issue at hand is addressed at PR #137. Let's move the discussion over there
